src/main/java/org/apache/maven/plugins/dependency/tree/VerboseGraphTextSerializer.java

```diff
@@ -45,8 +45,16 @@ public String serialize( DependencyNode root )
         // deal with root first
         Artifact rootArtifact = root.getArtifact();
         builder.append( rootArtifact.getGroupId() ).append( ":" ).append( rootArtifact.getArtifactId() ).append(
-                ":" ).append( rootArtifact.getExtension() ).append( ":" ).append( rootArtifact.getVersion() ).append(
-                "\n" );
+                ":" ).append( rootArtifact.getExtension() ).append( ":" ).append( rootArtifact.getVersion() );
+
+        if ( root.getData().containsKey( "ContainsModule" ) )
+        {
+            builder.append( " WARNING: this tree contains a submodule and once it reaches the submodule will print "
```

Comment:
I'd write:

contains a submodule; once it reaches

Reply:
or just convert to two sentences